cilium-health/launch/launcher.go

```diff
@@ -60,44 +89,59 @@ func (ch *CiliumHealth) Run() {
 		time.Sleep(connectRetryInterval)
 	}
 
+	// When the unix socket is made available, set its permissions.
+	os.Remove(defaults.SockPath)
+	go func() {
```

Comment:
nit: I'm not sure that this logic needs to be in a separate goroutine now, this could be run after the goroutine is launched below to serve the health API... since this will block until the socket is available, then set the permissions properly.

Reply:
Ohhh, good catch! I was already a bit unhappy with the fact that this code would basically never succeed on the first try (since the server likely hasn't started yet). Moving the code *after* we spawn the server goroutine should at least increase the likelihood that the socket will be there.